cmd-line-utils/readline/history.c

```diff
@@ -211,14 +211,14 @@ history_get (offset)
 
 HIST_ENTRY *
 alloc_history_entry (string, ts)
-     char *string;
+     const char *string;
      char *ts;
 {
   HIST_ENTRY *temp;
 
   temp = (HIST_ENTRY *)xmalloc (sizeof (HIST_ENTRY));
 
-  temp->line = string ? savestring (string) : string;
+  temp->line = string ? savestring (string) : NULL;
```

Comment:
What was the warning here?

Reply:
This is part of the other change in this file: string was a char*, and not a const char*. And line is simply a char* (and has to stay that), so we can't assign string directly to it, even when we know it's null.